### PR TITLE
grpc: Upgrade to v3

### PR DESCRIPTION
**Pull request by sirspinach**
This commit also ignores newly reported transient errors in foreman
that occur when quilt tries to contact a machine before the minion
container has started.

### CODE REVIEW

File: cluster/foreman.go [hunk 1]
```diff
@@ -202,7 +203,8 @@ func (c clientImpl) getMinion() (pb.MinionConfig, error) {
 	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
 	cfg, err := c.GetMinionConfig(ctx, &pb.Request{})
 	if err != nil {
-		if ctx.Err() == nil {
+		if ctx.Err() == nil && !strings.Contains(err.Error(),
+			"transport failure") {
```
Comment (mmussomele) [inline cluster/foreman.go]: should this be a separate patch?

Comment (sirspinach) [inline cluster/foreman.go]: Do error messages make the tester fail? I'm not sure if the commit message made this clear, but these errors weren't reported until gRPC v3

**Comment by sirspinach** [comment]
reworded summary

**Comment by ejj** [comment]
Merged.  Note that the commit message should have been wrapped to 70 characters.
